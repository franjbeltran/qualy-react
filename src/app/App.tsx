import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import './App.css';

import Home from '../pages/home/home.component';
import Login from '../pages/login/login.component';

import Header from '../components/header/header.component';

import { MainContainer } from './App.styles';

import { UserProvider } from '../contexts/user';

const router = createBrowserRouter([
  {
    path: '/',
    element: <Home />
  },
  {
    path: '/login',
    element: <Login />
  }
]);

const App = () => {
  return (
    <MainContainer>
      <UserProvider>
        <Header />
        <RouterProvider router={router} />
      </UserProvider>
    </MainContainer>
  );
}

export default App;
