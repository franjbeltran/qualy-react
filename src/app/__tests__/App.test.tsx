import { cleanup, render, RenderResult } from '@testing-library/react';
import App from '../App';

jest.mock('axios', () => ({
  axios: {
    get: jest.fn()
  }
}));

jest.mock('../../components/header/header.component', () => (
  (props: object) => {
    const Header = 'header';
    return <Header is='header' data-testid='header' {...props} />;
  }
));

jest.mock('react-router-dom', () => ({
  createBrowserRouter: jest.fn(),
  RouterProvider: (props: object) => {
    const RouterProvider = 'RouterProvider';
    return <RouterProvider is='router-provider' {...props} />
  }
}));

describe('App component', () => {
  afterEach(cleanup);

  it('renders the structure', () => {
    // Given

    // When
    const { container }: RenderResult = render(<App />);

    // Then
    expect(container).toMatchSnapshot();
  });
});
