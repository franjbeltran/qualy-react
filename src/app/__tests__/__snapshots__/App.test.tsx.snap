// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the structure 1`] = `
<div>
  <div
    class="sc-gKPRtg ioznsx"
  >
    <header
      data-testid="header"
      is="header"
    />
    <routerprovider
      is="router-provider"
    />
  </div>
</div>
`;
